src/sentry/tagstore/base.py

```diff
@@ -233,9 +232,9 @@ def get_group_tag_values(self, group_ids, environment_id, keys=None, values=None
         """
         raise NotImplementedError
 
-    def delete_tag_key(self, project_id, key):
+    def delete_tag_key(self, project_id, environment_id, key):
```

Comment:
It might be worth considering making delete operations take an optional environment instead of requiring one, and deleting from all environments if none is provided.

Reply:
Ya, I had a hard time with this because we (AFAIK) don't have any idea what the frontend will look like still. We have a "tags management" page and I guess people will be able to filter it by environment? But it seems like by default they wouldn't care, and they'd just want to nuke an entire key across all envs -- so you're right, will change this.